### ISSUE TITLE
Stuck on Cuda

### ISSUE THREAD

**Issue by OmidGhotbi** [taichi-dev/taichi_elements#108]
Most of the code will stuck after a few frames of simulation on the Cuda backend, I tested all of them on the CPU with no issue. other information you may need.
Platform Windows 11 pro x64
CPU intel 13900k
Gpu 3070 RTX 8GB
RAM DDR4 128 GB
Asus z790

**Comment by tsak1998** [comment]
Hello @OmidGhotbi. Did you had any luck overcoming this issue?

**Comment by OmidGhotbi** [comment]
> Hello @OmidGhotbi. Did you had any luck overcoming this issue?

Yes, this was an error caused by the 13th and 14th-generation Intel CPU issues that caused memory violation in VRAM, so it was not directly about Taichi. As soon as I changed my CPU the problem was mostly gone. 
But in the end, I stopped developing the code in Taichi for Blender and other software because the Taichi core is not designed to be used in another environment, for example after running any simulation it will not release the simulation and report the end of the simulation correctly so you will be force to close Bender and restart it to do another sim.

**Comment by YuanSha0** [comment]
@OmidGhotbi  Hi, my CPU is an Intel Xeon Platinum 8538C (Sapphire Rapids), so it’s not one of the affected 13th/14th Gen desktop CPUs. However, I’m still encountering the same issue. Are there any other known causes or possible workarounds?
